ext/liquid_c/tokenizer.c

```diff
@@ -189,9 +195,11 @@ static void tokenizer_next_for_template(tokenizer_t *tokenizer, token_t *token)
 
     if (token->type == TOKEN_VARIABLE || token->type == TOKEN_TAG) {
         token->str_trimmed += 2 + token->lstrip;
-        token->len_trimmed -= 2 + token->lstrip + 2 + token->rstrip;
+        token->len_trimmed -= 4 + (has_body ? token->lstrip + token->rstrip : (token->lstrip | token->rstrip));
```

Comment:
We don't need `has_body`, we just need to make sure we aren't decrementing len_trimmed below 0 here.

```suggestion
        token->len_trimmed -= 2 + token->lstrip + 2;
        if (token->rstrip && token->len_trimmed)
          token->len_trimmed--;
```